src/server/system_services/bucket_server.js

```diff
@@ -251,6 +252,49 @@ function validate_pool_constraints({ mongo_pool, default_pool }) {
     }
 }
 
+function validate_replication(req, configured_buckets, min_required) {
```

Comment:
I'm not a fan of validation functions that do other stuff. I think it's better to split it into 2 functions, `validate_replication` and `noramalize_replication_policy` or something

Reply:
splitted